### ISSUE TITLE
nixos-rebuild for nixos-21.11: `boot.kernel.sysctl.net.core.rmem_max` is defined multiple times

### ISSUE THREAD

**Issue by drperceptron** [NixOS/nixpkgs#151512]
### Describe the bug
Cannot build `nixos-21.11` if both `services.ipfs.enable` and `services.transmission.enable`.

### Steps To Reproduce
1. `configuration.nix`:
```
{ config, pkgs, lib, ... }:

{
  boot.loader.grub.device = "nodev";
  fileSystems."/".device = "/dev/null";
  services.ipfs.enable = true;
  services.transmission.enable = true;
}
```
2. Add `nixos-21.11` channel:
```
nix-channel --add https://nixos.org/channels/nixos-21.11 nixos
nix-channel --update
```
3. Build: `NIXOS_CONFIG=$(pwd)/configuration.nix nixos-rebuild dry-build`. Fails with error:
```
error: The unique option `boot.kernel.sysctl.net.core.rmem_max' is defined multiple times. Definition values:
- In `/nix/var/nix/profiles/per-user/root/channels/nixos/nixos/modules/services/torrent/transmission.nix': "4194304"
- In `/nix/var/nix/profiles/per-user/root/channels/nixos/nixos/modules/services/network-filesystems/ipfs.nix': 2500000
(use '--show-trace' to show detailed location information)
```

### Expected behavior
NixOS builds without error.

### Notify maintainers
@ju1m (transmission), ipfs has no maintainers listed

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
```
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.84, NixOS, 21.05.4547.2949ed36539 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.16`
 - channels(asdf): `""`
 - channels(root): `"nixos-21.11.334625.86453059bf8"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module: ipfs transmission
```

**Comment by ju1m** [comment]
@drperceptron, both `ipfs` and `transmission` are already using a `mkDefault` so I don't see what can be done, for me the user has a choice to make in this case and set `boot.kernel.sysctl."net.core.rmem_max"` to his/her liking.

**Comment by drperceptron** [comment]
@ju1m I have followed your advice and it corrected the reported issue. Thank you.

I attempted to take the initiative and figure out more about how `mkDefault` works. We could possibly change it mean something like "the maximum of this `mkDefault` and any previous `mkDefault` values". (Maximum works here but presumably isn't an appropriate policy for all options.) I was entirely out of my depth. Let's consider this closed unless somebody better informed than myself has strong opinions on the matter/better knowledge of the code base.

**Comment by drperceptron** [comment]
@ju1m I have followed your instruction and it corrected the reported issue. Building a system with both services enabled works (21.11). Thank you.

I attempted to take the initiative and figure out more about how mkDefault works. We could possibly change mkDefault/how it is evaluated to mean something like "the maximum of this mkDefault and any previous mkDefault values." (Maximum works here but presumably isn't an appropriate policy for all options.) I was entirely out of my depth. Let's consider this closed unless somebody better informed than myself has strong opinions on the matter/better knowledge of the code base.
